near/omni-bridge/src/storage.rs

```diff
@@ -183,6 +183,44 @@ impl Contract {
         }
     }
 
+    // Deducts `native_fee` from `account_id` and credits the remaining balance to `signer_id` and verify the `signer_id` has enough balance.
+    // Returns an error if `account_id` does not have enough balance to cover `native_fee` or if `signer_id` is not registered.
+    pub(crate) fn try_pay_native_fee_from_message_account(
+        &mut self,
+        account_id: &AccountId,
+        native_fee: u128,
+        storage_payer: &AccountId,
+        required_storage_payer_balance: NearToken,
+    ) -> Result<(), String> {
+        let balance = self
+            .accounts_balances
+            .get(account_id)
+            .ok_or("ERR_ACCOUNT_NOT_REGISTERED")?;
```

Comment:
Why is this an error? If we don’t have a native fee, then we don’t need to register an account.